test/test_dtype.py

```diff
@@ -85,7 +85,9 @@ def test_bf16_disk_write_read(self):
 @unittest.skipIf((getenv("CI", "") != "" and Device.DEFAULT in ["LLVM"]) or Device.DEFAULT == "WEBGPU", "float16 broken in some CI backends")
 class TestHalfDtype(unittest.TestCase):
   def test_float16_to_np(self): _test_to_np(Tensor([1,2,3,4], dtype=dtypes.float16), np.float16, [1,2,3,4])
+  @unittest.skipIf(Device.DEFAULT == "METAL", "Metal Backend does not support float16")
```

Comment:
Metal does support float16???

Reply:
Yup that's the miscompilation I was talking about, check out #1330 for more info.